evm-subql/src/mappings/mappingHandlers.ts

```diff
@@ -88,24 +84,31 @@ export async function handleEvmEvent(event: SubstrateEvent): Promise<void> {
 
   await transactionReceipt.save();
 
-  const existedIdx = (await Log.getByReceiptId(receiptId)).length;
-
   for (const [idx, evmLog] of ret.logs.entries()) {
     const log = Log.create({
-      id: `${receiptId}-${idx + existedIdx}`,
+      id: `${receiptId}-${idx}`,
       transactionHash,
-      blockNumber: block.block.header.number.toBigInt(),
-      blockHash: block.block.hash.toHex(),
-      transactionIndex,
+      blockNumber,
+      blockHash,
+      transactionIndex: BigInt(transactionIndex),
       removed: evmLog.removed,
       address: evmLog.address,
       data: evmLog.data,
       topics: evmLog.topics,
-      logIndex: BigInt(idx + existedIdx),
+      logIndex: BigInt(idx),
       receiptId,
       ...transactionInfo
     });
 
     await log.save();
   }
 }
+
+export async function handleBlock(block: SubstrateBlock): Promise<void> {
+  const TARGET_MODULES = ['EVM', 'DEX'];
+  const targetExtrinsics = block.block.extrinsics.filter((e) =>
+    TARGET_MODULES.includes(e.method.section.toUpperCase())
+  );
```

Comment:
We still need to filter the events of evm module. `honzon-bridge`/`currencies`/`load` and other modules will include the evm tx.

Reply:
use `findEvmEvent` to filter the EVM tx